src/main/java/hudson/plugins/jira/JiraSite.java

```diff
@@ -164,12 +177,20 @@ public JiraSite(URL url, URL alternativeUrl, String userName, String password, b
             }
 
         this.url = url;
+        
+        if(timeout != null) {
+        	this.timeout = timeout;
+        } else {
+        	this.timeout = JiraSite.DEFAULT_TIMEOUT;
```

Comment:
if u use dbsetter, no need to precheck for null. Simply use default value in constructor (and make field not final).